engine/uState.ml

```diff
@@ -287,14 +288,23 @@ let process_universe_constraints uctx cstrs =
         | UAlgebraic _ ->
           (* l contains a +1 and r=r' small so l <= r impossible *)
           sort_inconsistency Le l r
-        | USmall l' | ULevel l' ->
+        | USmall l' ->
+          if UGraph.check_leq_sort univs l r then local
+          else sort_inconsistency Le l r
+        | ULevel l' ->
           if UGraph.check_leq_sort univs l r then
+            (* This is only true when type-in-type... Should we keep it? *)
```

Comment:
Can't it happen with Prop <= Set too?

Reply:
But isn't Prop and Set restricted to the USmall case now?

Reply:
right this is `l <= Set`
We could still have `l = Set` in the graph though (equalities in the graph are not always reflected in the substitution)

Reply:
I suspect that the current code in master is already written with the assumption that graph equality and substitutions are synced, even if it's not true in practice. This piece of code really is a mess.

Reply:
I'm pretty sure equalities between rigid universes aren't in the substitution.

Reply:
Seems to work, though.